### DISCUSSION TITLE
MLC monitoring using Beamnrc

### DISCUSSION THREAD

**Discussion by Wanggl180328** [nrc-cnrc/EGSnrc#1181] [Questions]
I tried to use Beamnrc to track the behaviors of the MLC. I firstly tried to use the VARMLC codes provided by the manual to take place the second collimator, but the beam_gui show the data is mismatch. I think the version of the manual maybe too old. Therefore, I tried to  customize the VARMLC but I did not find somewhere to set the leaf parameters including the size of tongue and groove, the leaf gap or the leaf radius. Otherwise, I wonder if there is a component to describe a MLC with an arbitrary shape.
![捕获](https://github.com/user-attachments/assets/181923be-14c6-4d9f-8000-99baebee4714)

**Comment by rtownson** [comment]
The modern varian MLC models best fit the `SYNCVMLC` or `SYNCHDMLC` CMs in BEAMnrc. For those component modules you'll find detailed control of the MLC shape. There is no option for arbitrary shape, you would have to do that using egs++ instead of BEAMnrc, but it would be tedious/tricky.

**Reply by Wanggl180328** [reply]
OK I will try to use the egs++ to find if it works. But I found the egs++ may be difficult to get the PDD or beam profile. If I want to get these parameters, should I use the geometry to partition the plane I concerned?

**Reply by rtownson** [reply]
You could try asking Varian for the CAD mesh of their MLCs, and then use the new mesh support in egs++ to model the MLCs. So far no one has tried this yet. 

To get PDDs and profiles, you can use an XYZ geometry in egs++, and even an egsphant file in the same format that DOSXYZnrc uses. It's possible to output a 3ddose file from those and use the usual dose analysis tools. Otherwise, egs++ simulations can output the dose to every region, and so long as you understand the numbering you can produce a profile from that.

https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/classEGS__XYZGeometry.html

**Reply by Wanggl180328** [reply]
Thank you for your valuable advise.

**Reply by Wanggl180328** [reply]
I tried to use SYNCHDMLC CM, but when I tried to beam_build the related modules, many warnings appeared. When I opened the gui, the leaf details are also not availiable to adjust. These warning do not exist when I use other jaws.
![捕获](https://github.com/user-attachments/assets/2bcc6b1a-d436-4bdc-b88b-bfd925faf48a)

**Reply by rtownson** [reply]
If you're using SYNCHDMLC, why is it complaining about MLC_cm? Do you also have a MLC CM in the .module file? Make sure you completely remove the accelerator directory if you're trying to beam_build it with the same name after changing the .module file - or make sure to give the .module file a new name after making changes.
